### ISSUE TITLE
Running fin restart service disconnects all containers

### ISSUE THREAD

**Issue by sean-e-dietrich** [docksal/docksal#817]
When trying to restart one service example `fin restart cli` or `fin restart web` it disconnects all of the containers from the network. So you actually need to run a run a full `fin restart` on the project.

```
➜  golden_gate_national_parks_conservancy git:(master) ✗ fin start
Starting services...
Starting golden_gate_national_parks_conservancy_redis_1 ... done
Starting golden_gate_national_parks_conservancy_cli_1   ... done
Starting golden_gate_national_parks_conservancy_db_1    ... done
Starting golden_gate_national_parks_conservancy_web_1   ... done
Waiting for golden_gate_national_parks_conservancy_cli_1 to become ready...
Virtual Host: http://ggnpc.docksal

➜  golden_gate_national_parks_conservancy git:(master) ✗ fin status
                     Name                                   Command                  State                 Ports
---------------------------------------------------------------------------------------------------------------------------
golden_gate_national_parks_conservancy_cli_1     /opt/startup.sh supervisord      Up (healthy)   22/tcp, 3000/tcp, 9000/tcp
golden_gate_national_parks_conservancy_db_1      /entrypoint.sh mysqld            Up             0.0.0.0:32768->3306/tcp
golden_gate_national_parks_conservancy_redis_1   /docker-entrypoint.sh redi ...   Up             6379/tcp
golden_gate_national_parks_conservancy_web_1     httpd-foreground                 Up             443/tcp, 80/tcp

➜  golden_gate_national_parks_conservancy git:(master) ✗ fin restart web
Disconnecting project network...
Stopping services...
Stopping golden_gate_national_parks_conservancy_web_1 ... done
Starting services...
Creating network "golden_gate_national_parks_conservancy_default" with the default driver
golden_gate_national_parks_conservancy_db_1 is up-to-date
golden_gate_national_parks_conservancy_redis_1 is up-to-date
golden_gate_national_parks_conservancy_cli_1 is up-to-date
Starting golden_gate_national_parks_conservancy_web_1 ... done
Connected vhost-proxy to "golden_gate_national_parks_conservancy_default" network.

➜  golden_gate_national_parks_conservancy git:(master) ✗ fin status
                     Name                                   Command                  State            Ports
----------------------------------------------------------------------------------------------------------------
golden_gate_national_parks_conservancy_cli_1     /opt/startup.sh supervisord      Up (healthy)
golden_gate_national_parks_conservancy_db_1      /entrypoint.sh mysqld            Up
golden_gate_national_parks_conservancy_redis_1   /docker-entrypoint.sh redi ...   Up
golden_gate_national_parks_conservancy_web_1     httpd-foreground                 Up             443/tcp, 80/tcp

➜  golden_gate_national_parks_conservancy git:(master) ✗ fin restart
Disconnecting project network...
Stopping services...
Stopping golden_gate_national_parks_conservancy_web_1   ... done
Stopping golden_gate_national_parks_conservancy_cli_1   ... done
Stopping golden_gate_national_parks_conservancy_db_1    ... done
Stopping golden_gate_national_parks_conservancy_redis_1 ... done
Starting services...
Creating network "golden_gate_national_parks_conservancy_default" with the default driver
Starting golden_gate_national_parks_conservancy_cli_1   ... done
Starting golden_gate_national_parks_conservancy_db_1    ... done
Starting golden_gate_national_parks_conservancy_redis_1 ... done
Starting golden_gate_national_parks_conservancy_web_1   ... done
Waiting for golden_gate_national_parks_conservancy_cli_1 to become ready...
Connected vhost-proxy to "golden_gate_national_parks_conservancy_default" network.

➜  golden_gate_national_parks_conservancy git:(master) ✗ fin status
                     Name                                   Command                  State                 Ports
---------------------------------------------------------------------------------------------------------------------------
golden_gate_national_parks_conservancy_cli_1     /opt/startup.sh supervisord      Up (healthy)   22/tcp, 3000/tcp, 9000/tcp
golden_gate_national_parks_conservancy_db_1      /entrypoint.sh mysqld            Up             0.0.0.0:32769->3306/tcp
golden_gate_national_parks_conservancy_redis_1   /docker-entrypoint.sh redi ...   Up             6379/tcp
golden_gate_national_parks_conservancy_web_1     httpd-foreground                 Up             443/tcp, 80/tcp
```

**Comment by achekulaev** [comment]
Introduced in https://github.com/docksal/docksal/commit/a663d9a9598c7b7abb8042797501248f17c30c4c

**Comment by lmakarov** [comment]
@sean-e-dietrich as an immediate workaround you can use `fin dc restart [<service>]`.

I propose a partial fix for this via #823.
Once merged, `fin p restart [<service>]` (`fin p restart [<service>]`) will do the same.

Long term, `fin restart` should be refactored to only be used with individual containers and not projects (project services).

**Comment by achekulaev** [comment]
`dc restart` cannot be used in place of our restart routine. Instead e should make our restart routine not to disconnect containers.

**Comment by lmakarov** [comment]
@achekulaev ideally, start should reconnect containers to the network, but I don't really want to implement this functionality right now. If there is no easy and low risk fix for this, then this issue should be pushed out to the next release.

**Comment by achekulaev** [comment]
@lmakarov an easy solution if to only disconnect the network when "_stop_containers" has "--all" switch or no parameters.

**Comment by lmakarov** [comment]
Fixed via #827
